File: testdata/typescript/handlers.ts
import { MAX_RETRIES, API_URL } from './constants';
import type { Config, Logger } from './constants';
import { validateInput } from './utils';

export class PaymentHandler {
  private config: Config;

  constructor(config: Config) {
    this.config = config;
  }

  validate(data: unknown): boolean {
    return validateInput(data) && typeof data === 'object';
  }

  async processPayment(request: Record<string, unknown>): Promise<void> {
    for (let i = 0; i < MAX_RETRIES; i++) {
      if (this.validate(request)) {
        await fetch(API_URL);
        return;
      }
    }
  }
}

export function handleRefund(id: string): void {
  console.log(`Refund: ${id}`);
}

File: testdata/python/handlers.py
from .constants import TRUST_HINTS, MAX_RETRIES
from .utils import validate_input


class PaymentHandler:
    def __init__(self, config):
        self.config = config

    def _validate(self, data):
        return validate_input(data) and data.get("amount", 0) > 0

    def process_payment(self, request):
        if not self._validate(request):
            return None
        hints = TRUST_HINTS.get(request["type"])
        return {"status": "ok", "hints": hints}


class RefundHandler:
    def _validate(self, data):
        return data.get("refund_id") is not None

    def process_refund(self, request):
        for _ in range(MAX_RETRIES):
            if self._validate(request):
                return {"status": "refunded"}
        return {"status": "failed"}
